### PR TITLE
Fallback matcher must not match slashes

### PR DESCRIPTION
**Pull request by imod**
fixes https://github.com/microcks/microcks/issues/614

### CODE REVIEW

**Comment by imod** [comment]
@lbroudoux I don't understand why the tests fail in this repo, but work on my fork for the exact same commit: https://github.com/imod/microcks/actions/runs/2476011578

**Comment by lbroudoux** [comment]
Thanks for the PR! Seems like it fails on my machine too... Will try to investigate.

**Comment by lbroudoux** [comment]
I think that failure is just due to your test being too deterministic ...
When no precise response is found, it defaults to first matching operation and mime-type. But 'first' is not precise in MongoDB terms. Depending on how the import was done and Mongo internals (or version, or arch), results may vary. The important thing is that you receive a response coming from the details operation with `details` in the description. I think the integration test should be simplified just to check that criterion.

File: webapp/src/test/java/io/github/microcks/web/RestControllerIT.java [hunk 1]
```diff
@@ -88,4 +88,46 @@ public void testSwaggerMocking() {
...
+         JSONAssert.assertEquals("{"
+         		+ "		\"name\":\"Millefeuille\","
+         		+ "		\"description\":\"Detail - Delicieux Millefeuille pas calorique du tout\","
+         		+ "		\"size\":\"L\","
+         		+ "		\"price\":4.4,"
+         		+ "		\"status\":\"available\","
+         		+ "		\"street\":\"freestreet 3\","
+         		+ "		\"city\":\"Paris\""
+         		+ "}",
+               response.getBody(), JSONCompareMode.LENIENT);
```
Comment (lbroudoux) [inline webapp/src/test/java/io/github/microcks/web/RestControllerIT.java:128]: I think this part should be simplified just to check that we have a response coming from the `details` operation. Something like `description.startsWith("Detail -")` should be enough.

File: webapp/src/main/java/io/github/microcks/web/RestController.java [hunk 2]
```diff
@@ -143,7 +143,7 @@ public ResponseEntity<?> execute(
                if (operation.getResourcePaths() != null) {
                   // Produce a matching regexp removing {part} and :part from pattern.
                   String operationPattern = getURIPattern(operation.getName());
-                  operationPattern = operationPattern.replaceAll("\\{.+\\}", "(.)+");
+                  operationPattern = operationPattern.replaceAll("\\{.+\\}", "([^/])+");
```
Comment (lbroudoux) [inline webapp/src/main/java/io/github/microcks/web/RestController.java:146]: Shouldn't we have something similar on next line too?

**Review by lbroudoux — CHANGES_REQUESTED** [review]
Need to make the test more robust by simplifying assertions.

**Comment by imod** [comment]
hmm, not sure I follow...
I think there is only one example of `details` imported at all, therefore I also expect this one to be returned - but maybe I don't understand it completely

File: webapp/src/main/java/io/github/microcks/web/RestController.java (hunk 2 above)
Comment (imod) [reply webapp/src/main/java/io/github/microcks/web/RestController.java:146]: yeah, thats probably true

**Comment by lbroudoux** [comment]
Actually I see 2 examples for the `details` responses in the OpenAPI file you attached to integration test. One for `Millefeuille` and another for `Eclair Cafe`...

**Comment by imod** [comment]
@lbroudoux I simplified the test and adjusted the pattern for the second case too. The test are fine on my branch again, but fail on this one with an error which is related to a problem of the infrastructure. Maybe you can trigger the action again?
https://github.com/microcks/microcks/actions/runs/2487395986

**Comment by imod** [comment]
@lbroudoux I resolved all the open points

File: webapp/src/test/java/io/github/microcks/web/RestControllerIT.java [hunk 3]
```diff
@@ -88,4 +91,32 @@ public void testSwaggerMocking() {
...
+   @Test
+   public void testFallbackMatchingWithRegex() {
+      // Upload modified pastry spec
+      uploadArtifactFile("target/test-classes/io/github/microcks/util/openapi/pastry-with-details-openapi.yaml", true);
+
+      // Check operation with a defined mock (name: 'Millefeuille')
+      ResponseEntity<String> response = restTemplate.getForEntity("/rest/pastry-details/1.0.0/pastry/Millefeuille/details", String.class);
+      assertEquals(200, response.getStatusCode().value());
+      try {
+    	  JSONObject details = new JSONObject(response.getBody());
```
Comment (lbroudoux) [inline webapp/src/test/java/io/github/microcks/web/RestControllerIT.java:104]: LGTM but code style has extra indent on 2nd and 3rd lines

File: webapp/src/main/java/io/github/microcks/web/RestController.java [hunk 4]
```diff
@@ -143,8 +143,8 @@ public ResponseEntity<?> execute(
                if (operation.getResourcePaths() != null) {
                   // Produce a matching regexp removing {part} and :part from pattern.
                   String operationPattern = getURIPattern(operation.getName());
-                  operationPattern = operationPattern.replaceAll("\\{.+\\}", "(.)+");
-                  operationPattern = operationPattern.replaceAll("(/:[^:^/]+)", "\\/(.+)");
+                  operationPattern = operationPattern.replaceAll("\\{.+\\}", "([^/])+");
+                  operationPattern = operationPattern.replaceAll("(/:[^:^/]+)", "([^/])+");
```
Comment (lbroudoux) [inline webapp/src/main/java/io/github/microcks/web/RestController.java:147]: I think this one is not OK as it removes first /.
`operationPattern = operationPattern.replaceAll("(/:[^:^/]+)", "\\/([^/]+)");` is the way to go. COuld you please modify this one ?

**Review by lbroudoux — CHANGES_REQUESTED** [review]
Need a fix on postman notation and remove extra indents and we're good to merge!

File: webapp/src/test/java/io/github/microcks/web/RestControllerIT.java [hunk 5]
```diff
@@ -88,4 +91,32 @@ public void testSwaggerMocking() {
...
+
+      // Check operation with a defined mock (name: 'Millefeuille')
+      ResponseEntity<String> response = restTemplate.getForEntity("/rest/pastry-details/1.0.0/pastry/Millefeuille/details", String.class);
+      assertEquals(200, response.getStatusCode().value());
+      try {
+         JSONObject details = new JSONObject(response.getBody());
+         String description = details.getString("description");
+         assertTrue(description.startsWith("Detail -"));   
+      } catch (Exception e) {
+    	 fail("No Exception should be thrown here");
```
Comment (imod) [inline webapp/src/test/java/io/github/microcks/web/RestControllerIT.java:108]: I don't know whats happening with these indents - copy paste just does not work on my site
Do you have a code formatter rule or anything I can use to apply?

**Comment by imod** [comment]
sorry, this took a lot more commits then it should - some how the formatting was not displayed the same way on GH as in my IDE...

**Review by lbroudoux — APPROVED** [review]
Thanks!

**Pull request merged**